packages/test/test-end-to-end-tests/src/test/summarizeRefreshLatestSummary.spec.ts

```diff
@@ -53,5 +48,79 @@ describeCompat(
 			await createSummarizer(provider, container, undefined, summaryVersion);
 			await provider.ensureSynchronized();
 		});
+
+		itExpects(
+			"The summarizing client will immediately refresh its own summaries",
+			[
+				{ eventName: "fluid:telemetry:Summarizer:Running:Summarize_start" },
+				{ eventName: "fluid:telemetry:Summarizer:Running:Summarize_end" },
+				{ eventName: "fluid:telemetry:SummarizerNode:refreshLatestSummary_start" },
+				{ eventName: "fluid:telemetry:SummarizerNode:refreshLatestSummary_end" },
+				{ eventName: "fluid:telemetry:Summarizer:Running:Summarize_start" },
+				{ eventName: "fluid:telemetry:Summarizer:Running:Summarize_end" },
+				{ eventName: "fluid:telemetry:SummarizerNode:refreshLatestSummary_start" },
+				{ eventName: "fluid:telemetry:SummarizerNode:refreshLatestSummary_end" },
+				{ eventName: "fluid:telemetry:Summarizer:Running:Summarize_start" },
+				{ eventName: "fluid:telemetry:Summarizer:Running:Summarize_end" },
+				{ eventName: "fluid:telemetry:SummarizerNode:refreshLatestSummary_start" },
+				{ eventName: "fluid:telemetry:SummarizerNode:refreshLatestSummary_end" },
+			],
+			async () => {
+				const summaryConfig: ISummaryConfiguration = {
+					...DefaultSummaryConfiguration,
+					...{
+						maxTime: 5000 * 12,
+						maxAckWaitTime: 120000,
+						maxOps: 1,
+						initialSummarizerDelayMs: 0,
+					},
+				};
+				const testContainerConfig: ITestContainerConfig = {
+					runtimeOptions: {
+						summaryOptions: {
+							summaryConfigOverrides: summaryConfig,
+						},
+					},
+					loaderProps: {
+						configProvider: mockConfigProvider({
+							"Fluid.Summarizer.immediatelyRefreshLatestSummaryAck": true,
+						}),
+					},
+				};
+				const container = await provider.makeTestContainer(testContainerConfig);
+				const container2 = await provider.loadTestContainer(testContainerConfig);
+				const dataObject = (await container.getEntryPoint()) as ITestDataObject;
+				const dataObject2 = (await container2.getEntryPoint()) as ITestDataObject;
+				await waitForContainerConnection(container);
+				await waitForContainerConnection(container2);
+
+				// The first summary needs to be generated otherwise the summarizer will only generate one summary.
+				dataObject._root.set(`first`, "op");
+				await provider.ensureSynchronized();
+
+				const array: string[] = [];
```

Comment:
What is this array doing? And why does container3 load from the first ack?

Reply:
To be honest, I was curious as to what the behavior of the container would be, whether or not the 2nd op would be summarized or not. Based on the tests, with the local server, the 2nd op would be summarized in 1st the summary. With the tinylicious server, I think the 2nd op would not be summarized until the 2nd summary. It's interesting behavior that I wasn't sure if it was important to note. The summaries generated would be different, but the document should recover/become eventually consistent. This is a rare edge case.

Reply:
Another note, the array should contain 2 acks, and I should check for that at the end of the test. Another note, if we update the summary heuristics runner in any fashion, this test will likely start failing.